plugins/modules/attachment.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2022, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = r"""
+module: attachment
+
+author:
+  - Polona Mihalič (@PolonaM)
+
+short_description: a module that users can use to download attachment using sys_id
+
+description:
+  - Download attachment using attachment's sys_id.
+version_added: 2.0.0
+extends_documentation_fragment:
+  - servicenow.itsm.instance
+
+options:
+  dest:
+    description:
+      - Specify the path in which the attachment will be downloaded to. The file will be downloaded on all the hosts from the inventory.
+    type: str
+    required: true
+  sys_id:
+    description:
+      - Attachment's sys_id.
+    type: str
+    required: true
+
+notes:
+  - Supports check_mode.
+"""
+
+EXAMPLES = """
+  - name: ServiceNow download attachment module
+    servicenow.itsm.attachment:
+      instance:
+        host: https://instance_id.service-now.com
+        username: user
+        password: pass
+      dest: /tmp/sn-attachment
+      sys_id: 003a3ef24ff1120031577d2ca310c74b
+"""
+
+
+RETURN = r"""
+record:
+  description: download attachment record
+  returned: success
+  type: dict
+  contains:
+    elapsed:
+        description: the number of seconds that elapsed while performing the download
+        returned: success
+        type: float
+        sample: 2.3
+    msg:
+        description: OK or error message
+        returned: always
+        type: str
+        sample: OK
+    size:
+        description: size of the target
+        returned: success
+        type: int
+        sample: 1220
+    status_code:
+        description: the HTTP status code from the request
+        returned: success
+        type: int
+        sample: 200
+"""
+
+
+import time
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils import (
+    arguments,
+    client,
+    attachment,
+    errors,
+)
+
+
+def run(module, attachment_client):
+    start = time.time()
+    response = attachment_client.get_attachment(module.params["sys_id"])
+    if response.status == 200:
+        if not module.check_mode:
+            attachment_client.save_attachment(response.data, module.params["dest"])
+    elif response.status == 404:
+        raise errors.ServiceNowError(
+            "Status code: 404, Details: " + json.loads(response.data)["error"]["detail"]
+        )
+    end = time.time()
+    elapsed = round(end - start, 2)
+    size = int(json.loads(response.headers["x-attachment-metadata"])["size_bytes"])
```

Comment:
This could result in KeyError or ValueError, which can be handled as a fallback by reading file size from the downloaded file.

Reply:
Implemented with the try-exept